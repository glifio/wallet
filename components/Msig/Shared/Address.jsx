import React from 'react'
import PropTypes from 'prop-types'
import {
  Box,
  Glyph,
  CopyAddress,
  Label,
  IconLedger,
  ButtonClose,
  ButtonEdit
} from '@glif/react-components'
import { ADDRESS_PROPTYPE } from '../../../customPropTypes'

const Address = ({
  widthOverride,
  address,
  label,
  glyphAcronym,
  onRemoveSigner,
  onChangeSigner
}) => {
  return (
    <Box
      display='flex'
      alignItems='center'
      justifyContent='flex-start'
      color='core.darkgray'
      bg='background.messageHistory'
      height={6}
      px={2}
      mr={2}
      my={1}
      borderRadius={2}
      maxWidth={widthOverride ? 'none' : 12}
      width={widthOverride || 'auto'}
      minWidth={11}
    >
      {glyphAcronym ? (
        <Glyph
          mr={3}
          color='core.nearblack'
          acronym={glyphAcronym}
          size={5}
          border={0}
        />
      ) : (
        <Glyph
          justifyContent='flex-end'
          alignSelf='flex-end'
          mb='1px'
          mr={3}
          size={5}
          Icon={IconLedger}
          color='core.nearblack'
          bg='transparent'
          fill='#444'
          border={0}
          css='transform:translateY(-6px)'
        />
      )}
      <Box flexGrow='1'>
        <Label fontSize={1}>{label}</Label>
        <CopyAddress color='core.nearblack' address={address} />
      </Box>
      {onChangeSigner && (
        <ButtonEdit
          role='button'
          type='button'
          onClick={onChangeSigner}
          stroke='core.darkgray'
          aria-label='edit-signer'
        />
      )}
      {onRemoveSigner && (
        <ButtonClose
          fill='core.darkgray'
          role='button'
          type='button'
          onClick={onRemoveSigner}
          aria-label='remove-signer'
        />
      )}
    </Box>
  )
}

Address.propTypes = {
  address: ADDRESS_PROPTYPE,
  label: PropTypes.string.isRequired,
  glyphAcronym: PropTypes.string,
  widthOverride: PropTypes.string,
  onRemoveSigner: PropTypes.func,
  onChangeSigner: PropTypes.func
}

export default Address
