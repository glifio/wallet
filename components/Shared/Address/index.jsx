import React, { forwardRef, useState } from 'react'
import styled from 'styled-components'
import { space, color, layout, border, flexbox } from 'styled-system'
import { ADDRESS_PROPTYPE } from '../../../customPropTypes'
import Box from '../Box'
import { Text } from '../Typography'
import theme from '../theme'
import copyToClipboard from '../../../utils/copyToClipboard'

const Wrapper = styled(Box).attrs(() => ({
  border: 1,
  borderColor: 'input.border',
  borderRadius: 1,
  pl: 3,
  width: 14
}))`
  display: flex;
  height: 70px;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  ${space};
`

const InlineButton = styled.button.attrs(() => ({
  border: 0,
  borderLeft: 1,
  borderColor: 'input.border',
  borderTopRightRadius: 1,
  borderBottomRightRadius: 1,
  bg: 'input.background.base',
  width: 7,
  p: 0,
  ...theme.textStyles.label
}))`
  display: inline-block;
  &:hover {
    cursor: pointer;
  }
  height: 100%;
  outline: none;
  ${color}
  ${space}
  ${layout}
  ${border}
  ${flexbox}
`

const Address = forwardRef(({ address, ...props }, ref) => {
  const [showAddress, setShowAddress] = useState(false)
  return (
    <Wrapper ref={ref} {...props}>
      {showAddress ? (
        <Text pl={1} pr={2}>
          {address}
        </Text>
      ) : (
        <Box pl={1} pr={2} display='flex' flexDirection='row'>
          <Text>{address.slice(0, 2)}</Text>
          <Text mx={1}>{Array(address.length - 10).join('‚óè')}</Text>
          <Text>{address.slice(-4)}</Text>
        </Box>
      )}

      <Box height='100%' display='flex' flexDirection='row'>
        <InlineButton onClick={() => copyToClipboard(address)}>
          Copy
        </InlineButton>
        <InlineButton onClick={() => setShowAddress(!showAddress)}>
          {showAddress ? 'Hide' : 'Reveal'}
        </InlineButton>
      </Box>
    </Wrapper>
  )
})

Address.propTypes = {
  /**
   * filecoin address
   */
  address: ADDRESS_PROPTYPE
}

Address.defaultProps = {
  address: ''
}

export default Address
