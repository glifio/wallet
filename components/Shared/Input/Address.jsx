import React, { forwardRef } from 'react'
import { func, string, bool } from 'prop-types'
import { validateAddressString } from '@openworklabs/filecoin-address'
import TextInput from './Text'

const Address = forwardRef(
  ({ value, label, error, setError, ...props }, ref) => {
    return (
      <TextInput
        onBlur={() => {
          const isValidAddress = validateAddressString(value)
          if (value && !isValidAddress) setError(`Invalid ${label} address.`)
        }}
        onFocus={() => {
          if (error) setError('')
        }}
        ref={ref}
        label={label}
        value={value}
        error={error}
        {...props}
      />
    )
  }
)

Address.propTypes = {
  onChange: func,
  label: string,
  setError: func,
  value: string,
  error: string,
  placeholder: string,
  valid: bool
}

Address.defaultProps = {
  value: '',
  placeholder: 'f1...',
  onChange: () => {},
  setError: () => {},
  label: ''
}

export default Address
