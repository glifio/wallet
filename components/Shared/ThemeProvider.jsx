import React from 'react'
import PropTypes from 'prop-types'
import { ThemeProvider as StyledThemeProvider } from 'styled-components'
import theme from './theme'

/* eslint-disable no-shadow */
const ThemeProvider = ({ children, theme, ...rest }) => {
  return (
    <StyledThemeProvider theme={theme} {...rest}>
      {children}
    </StyledThemeProvider>
  )
}

ThemeProvider.propTypes = {
  theme: PropTypes.object,
  children: PropTypes.node
}

ThemeProvider.defaultProps = {
  theme,
  children: <></>
}

ThemeProvider.displayName = 'ThemeProvider'

export default ThemeProvider
