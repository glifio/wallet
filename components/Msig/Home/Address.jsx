import React from 'react'
import PropTypes from 'prop-types'
import { ADDRESS_PROPTYPE } from '../../../customPropTypes'

import { Box, Glyph, CopyAddress, Label, IconLedger } from '../../Shared'

const Address = ({ address, label, glyphAcronym }) => {
  return (
    <Box
      display='flex'
      alignItems='center'
      justifyContent='flex-start'
      color='core.darkgray'
      bg='background.messageHistory'
      height={6}
      px={2}
      mr={2}
      my={1}
      borderRadius={2}
      maxWidth={11}
    >
      {glyphAcronym ? (
        <Glyph
          mr={3}
          color='core.nearblack'
          acronym={glyphAcronym}
          size={5}
          border={0}
        />
      ) : (
        <Glyph
          justifyContent='flex-end'
          alignSelf='flex-end'
          mb='1px'
          mr={3}
          size={5}
          Icon={IconLedger}
          color='core.nearblack'
          bg='transparent'
          fill='#444'
          border={0}
          css='transform:translateY(-6px)'
        />
      )}
      <Box flexGrow='1'>
        <Label fontSize={1}>{label}</Label>
        <CopyAddress
          justifyContent='space-between'
          color='core.nearblack'
          address={address}
        />
      </Box>
    </Box>
  )
}

Address.propTypes = {
  address: ADDRESS_PROPTYPE,
  label: PropTypes.string.isRequired,
  glyphAcronym: PropTypes.string
}

export default Address
