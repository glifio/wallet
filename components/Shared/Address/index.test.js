import React from 'react'
import { cleanup, render, act, fireEvent } from '@testing-library/react'
import Address from '.'
import noop from '../../../utils/noop'
import ThemeProvider from '../ThemeProvider'

describe('Receive', () => {
  afterEach(cleanup)

  test('it renders correctly', () => {
    const { container } = render(
      <ThemeProvider>
        <Address
          address='t137sjdbgunloi7couiy4l5nc7pd6k2jmq32vizpy'
          close={noop}
        />
      </ThemeProvider>
    )

    expect(container.firstChild).toMatchSnapshot()
  })

  test('it reveals the address when the user clicks reveal', () => {
    const { container, getByText } = render(
      <ThemeProvider>
        <Address
          address='t137sjdbgunloi7couiy4l5nc7pd6k2jmq32vizpy'
          close={noop}
        />
      </ThemeProvider>
    )

    act(() => {
      fireEvent.click(getByText('Reveal'))
    })

    expect(container.firstChild).toMatchSnapshot()
  })
})
