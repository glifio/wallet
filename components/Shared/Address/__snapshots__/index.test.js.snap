// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Receive it renders correctly 1`] = `
.c1 {
  min-width: 0;
  box-sizing: border-box;
  padding-left: 4px;
  padding-right: 8px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c4 {
  min-width: 0;
  box-sizing: border-box;
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c2 {
  font-size: 1.25rem;
  font-weight: 400;
  line-height: 1.4;
  font-family: RT-Alias-Grotesk;
}

.c3 {
  font-size: 1.25rem;
  font-weight: 400;
  line-height: 1.4;
  font-family: RT-Alias-Grotesk;
  margin-left: 4px;
  margin-right: 4px;
}

.c0 {
  min-width: 0;
  box-sizing: border-box;
  border: 1px solid;
  border-color: #999999;
  border-radius: 1px;
  padding-left: 16px;
  width: 640px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 70px;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-left: 16px;
}

.c5 {
  display: inline-block;
  height: 100%;
  outline: none;
  background-color: #E4EBFC;
  padding: 0;
  width: 64px;
  border: 0;
  border-left: 1px solid;
  border-color: #999999;
  border-top-right-radius: 1px;
  border-bottom-right-radius: 1px;
}

.c5:hover {
  cursor: pointer;
}

<div
  class="c0"
  width="14"
>
  <div
    class="c1"
    display="flex"
  >
    <p
      class="c2"
      font-family="RT-Alias-Grotesk"
      font-size="3"
      font-weight="400"
    >
      t1
    </p>
    <p
      class="c3"
      font-family="RT-Alias-Grotesk"
      font-size="3"
      font-weight="400"
    >
      ●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●
    </p>
    <p
      class="c2"
      font-family="RT-Alias-Grotesk"
      font-size="3"
      font-weight="400"
    >
      izpy
    </p>
  </div>
  <div
    class="c4"
    display="flex"
    height="100%"
  >
    <button
      class="c5"
      font-family="RT-Alias-Grotesk"
      font-size="2"
      font-weight="400"
      width="7"
    >
      Copy
    </button>
    <button
      class="c5"
      font-family="RT-Alias-Grotesk"
      font-size="2"
      font-weight="400"
      width="7"
    >
      Reveal
    </button>
  </div>
</div>
`;

exports[`Receive it reveals the address when the user clicks reveal 1`] = `
.c2 {
  min-width: 0;
  box-sizing: border-box;
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c1 {
  font-size: 1.25rem;
  font-weight: 400;
  line-height: 1.4;
  font-family: RT-Alias-Grotesk;
  padding-left: 4px;
  padding-right: 8px;
}

.c0 {
  min-width: 0;
  box-sizing: border-box;
  border: 1px solid;
  border-color: #999999;
  border-radius: 1px;
  padding-left: 16px;
  width: 640px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 70px;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-left: 16px;
}

.c3 {
  display: inline-block;
  height: 100%;
  outline: none;
  background-color: #E4EBFC;
  padding: 0;
  width: 64px;
  border: 0;
  border-left: 1px solid;
  border-color: #999999;
  border-top-right-radius: 1px;
  border-bottom-right-radius: 1px;
}

.c3:hover {
  cursor: pointer;
}

<div
  class="c0"
  width="14"
>
  <p
    class="c1"
    font-family="RT-Alias-Grotesk"
    font-size="3"
    font-weight="400"
  >
    t137sjdbgunloi7couiy4l5nc7pd6k2jmq32vizpy
  </p>
  <div
    class="c2"
    display="flex"
    height="100%"
  >
    <button
      class="c3"
      font-family="RT-Alias-Grotesk"
      font-size="2"
      font-weight="400"
      width="7"
    >
      Copy
    </button>
    <button
      class="c3"
      font-family="RT-Alias-Grotesk"
      font-size="2"
      font-weight="400"
      width="7"
    >
      Hide
    </button>
  </div>
</div>
`;
